src/libraries/Microsoft.XmlSerializer.Generator/src/GenerateXmlSerializers.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.XmlSerializer.Generator
+{
+    /// <summary>
+    /// MSBuild task to run the Microsoft.XmlSerializer.Generator directly,
+    /// without spawning a new process.
+    /// </summary>
+    public class GenerateXmlSerializers : Task
+    {
+        /// <summary>
+        /// Path to the target assembly (usually the output DLL).
+        /// </summary>
+        [Required]
+        public string AssemblyPath { get; set; }
+
+        /// <summary>
+        /// Optional path to a response file containing additional generator arguments.
+        /// </summary>
+        public string RspFilePath { get; set; }
+
+        /// <summary>
+        /// Force regeneration of serializers even if they exist.
+        /// </summary>
+        public bool Force { get; set; }
+
+        /// <summary>
+        /// Run in quiet mode with minimal output.
+        /// </summary>
+        public bool Quiet { get; set; }
+
+        /// <summary>
+        /// If true, errors during serializer generation will be suppressed.
+        /// </summary>
+        public bool IgnoreErrors { get; set; }
+
+        /// <summary>
+        /// Executes the XmlSerializer generator.
+        /// </summary>
+        public override bool Execute()
+        {
+            var args = new List<string>();
+
+            if (!string.IsNullOrEmpty(AssemblyPath))
+                args.Add(AssemblyPath);
+
+            if (Force)
+                args.Add("--force");
+
+            if (Quiet)
+                args.Add("--quiet");
```

Comment:
These two are always true and we should update Sgen to remove all code that would run if they were false.